### PR TITLE
fix(host_id): get host id from `nodetool info`

### PR DESCRIPTION
**Pull request by fruch**
get host id from `nodetool info` and not from our `nodetool status` implemetiontion which is much more complex since it's mapping the node based on IP, and we don't need that for getting the host_id

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
this affect the sni usage that uses the host_id to connect to specific nodes

**Review by soyacz — DISMISSED** [review]

File: sdcm/cluster.py [hunk 1]
```diff
@@ -368,11 +368,15 @@ def region(self):
 
     @property
     def host_id(self):
-        full_nodetool_status = self.parent_cluster.get_nodetool_status(verification_node=self)
-        for data_center in full_nodetool_status:
-            if self.ip_address in full_nodetool_status[data_center]:
-                return full_nodetool_status[data_center][self.ip_address]['host_id']
-        raise AssertionError(f"Could not find the requested node {self.ip_address} in nodetool status")
+        info = self.run_nodetool('info', publish_event=False)
```
Comment (vponomaryov) [inline sdcm/cluster.py:371]: I like the approach that is used in the existing `def get_nodetool_info(node)` method.
It is the approach where we transform whole output to python dict and then just pick up any key out of it.

So, as a result I would expect something like following:

```
class BaseNode(AutoSshContainerMixin, WebDriverContainerMixin):
    def get_nodetool_info(self, publish_event=True):
        res = self.run_nodetool('info', publish_event=publish_event)
        proper_yaml_output = "\n".join([line for line in res.stdout.splitlines() if ":" in line])
        info_res = yaml.safe_load(proper_yaml_output)
        return info_res

    @property
    def host_id(self):
        return self.get_nodetool_info(publish_event=False).get('ID')

    ...

class BaseScyllaCluster:
    @staticmethod
    def get_nodetool_info(node):
        return node.get_nodetool_info()

    ...
```

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:371]: I've actully copy pasted this code from CCM, didn't see we have implementation for it

Comment (fruch) [reply sdcm/cluster.py:371]: but your direction better

**Review by soyacz — APPROVED** [review]

**Pull request merged**